src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingQueryMessageWireProtocol.cs

```diff
@@ -222,7 +222,16 @@ private IBsonSerializer CreateSizeLimitingPayloadSerializer(Type1CommandMessageS
         private void IgnoreResponse(IConnection connection, QueryMessage message, CancellationToken cancellationToken)
         {
             var encoderSelector = new ReplyMessageEncoderSelector<IgnoredReply>(IgnoredReplySerializer.Instance);
-            connection.ReceiveMessageAsync(message.RequestId, encoderSelector, _messageEncoderSettings, cancellationToken).IgnoreExceptions();
+            try
+            {
+                connection.ReceiveMessageAsync(message.RequestId, encoderSelector, _messageEncoderSettings, cancellationToken)
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            catch
+            {
+                // ignored
+            }
```

Comment:
`IgnoreResponse` is a sync method and should be 100% sync without calling any async methods:

    private void IgnoreResponse(IConnection connection, QueryMessage message, CancellationToken cancellationToken)
    {
        var encoderSelector = new ReplyMessageEncoderSelector<IgnoredReply>(IgnoredReplySerializer.Instance);
        try
        {
            connection.ReceiveMessage(message.RequestId, encoderSelector, _messageEncoderSettings, cancellationToken);
        }
        catch
        {
            // ignore exceptions
        }
    }

Reply:
And you should add a 100% async version that can be called from `ExecuteAsync`:

    private async Task IgnoreResponseAsync(IConnection connection, QueryMessage message, CancellationToken cancellationToken)
    {
        var encoderSelector = new ReplyMessageEncoderSelector<IgnoredReply>(IgnoredReplySerializer.Instance);
        try
        {
            await connection.ReceiveMessageAsync(message.RequestId, encoderSelector, _messageEncoderSettings, cancellationToken).ConfigureAwait(false);
        }
        catch
        {
            // ignore exceptions
        }
    }

Reply:
Line 148 in `ExecuteAsync` should be:

    await IgnoreResponseAsync(connection, message, cancellationToken).ConfigureAwait(false);

Reply:
Great suggestion! I didn't realize there's an async version. Now the code looks much better.
Done.